users/engler/fairseq/training.py

```diff
@@ -183,25 +189,41 @@ def create_files(self):
-                    manifest_lines[0] = bundle_lines[0]
-                with open(f"{self.out_cached_audio_manifest.get_path()}/{name}", "w") as cached_audio_manifest_file:
-                    cached_audio_manifest_file.write('\n'.join(manifest_lines))
+            if self.zipped_audio_dir is None:
+                for name in ["train.tsv", "valid.tsv"]:
+                    with open(f"{manifest_path}/{name}", "r") as manifest_file:
+                        manifest_lines = manifest_file.read().splitlines()
+                    audio_path = manifest_lines[0]
+                    bundle_lines = map(lambda line: audio_path + "/" + line.split("\t")[0], manifest_lines[1:])
+                    with open(f"{name}.bundle", 'w') as bundle_file:
+                        bundle_file.write("\n".join(bundle_lines))
+                    try:
+                        cached_audio_fn = sp.check_output(["cf", f"{name}.bundle"]).strip().decode("utf8")
+                    except sp.CalledProcessError:
+                        print(f"Cache manager: Error occurred for files in {name}")
+                        raise
+
+                    with open(cached_audio_fn) as local_bundle:
+                        bundle_lines = list(map(os.path.dirname, local_bundle.readlines()))
+                        assert bundle_lines.count(bundle_lines[0]) == len(bundle_lines), f"not all {name} files in same directory"
+                        manifest_lines[0] = bundle_lines[0]
+                    with open(f"{self.out_cached_audio_manifest.get_path()}/{name}", "w") as cached_audio_manifest_file:
+                        cached_audio_manifest_file.write('\n'.join(manifest_lines))
+            else:   # zipped audio data is given and we cache and unzip the zip file instead
+                if self.use_cache_manager:
+                    try:
+                        cached_audio_zip_dir = sp.check_output(["cf", self.zipped_audio_dir]).strip().decode("utf8")
+                        local_unzipped_audio_dir = f"{os.path.dirname(cached_audio_zip_dir)}/audio"
+                        sp.check_call(["unzip", "-q", "-n", cached_audio_zip_dir, "-d", local_unzipped_audio_dir])
+                    except sp.CalledProcessError:
+                        print(f"Cache manager: Error occurred for caching and unzipping audio data")
+                        raise
+                    for name in ["train.tsv", "valid.tsv"]:
+                        with open(f"{manifest_path}/{name}", "r") as manifest_file:
+                            manifest_lines = manifest_file.read().splitlines()
+                        manifest_lines[0] = local_unzipped_audio_dir
+                        with open(f"{self.out_cached_audio_manifest.get_path()}/{name}", "w") as cached_audio_manifest_file:
+                            cached_audio_manifest_file.write('\n'.join(manifest_lines))
```

Comment:
There is currently no check that the audio files contained in the zip file match the ones in the manifest. Maybe this is OK for now, but we should add a warning in the docstring then.

Reply:
Such a check should be simple to do. However I would only check that all files mentioned in the manifest exist in the unzipped directory and not the other way around. Do you think we should also check that all files in the unzipped directory exist in the manifest?

Reply:
No this one way should be enough as you create the manifest that is actually used based on the original manifest. So additional files in the zip would just be ignored.